master/cmd/stream-gen/main.go

```diff
@@ -238,6 +240,100 @@ func (b *Builder) String() string {
 	return b.builder.String()
 }
 
+func genTypescript(streamables []Streamable) ([]byte, error) {
+	b := Builder{}
+	typeAnno := func(f Field) ([2]string, error) {
+		x := map[string]([2]string){
+			"JSONB":            {"any", "{}"},
+			"string":           {"string", ""},
+			"bool":             {"bool", "false"},
+			"int":              {"number", "0"},
+			"int64":            {"number", "0"},
+			"[]int":            {"Array<number>", "[]"},
+			"time.Time":        {"string", ""},
+			"*time.Time":       {"string | undefined", "undefined"},
+			"model.TaskID":     {"string", ""},
+			"model.RequestID":  {"number", "0"},
+			"*model.RequestID": {"number | undefined", "undefined"},
+			"model.State":      {"string", ""},
```

Comment:
shouldn't this be an enum?

Reply:
I modified this part to use enum

Reply:
sorry i wasn't clear -- shouldn't the generated typescript type be an enum?

Reply:
Do you mean we should use a enum like `CommandState` for `model.State`?

Reply:
Yes -- if the field has constraints, we can communicate that through the type and make sure that we're handling all cases.

Reply:
Got it, fixed!